### PR TITLE
#4185 Add event generation to devmode

### LINKED ISSUE #4185: Add an option to generate Events in DevMode
### Situation Description

Currently, the DevMode can be used to generate random cases and contacts for testing and showcase purposes. The same would be needed for events, as it can be quite a lot of work to generate events manually and then fill each with participants.

### Feature Description

Add the same for events, with an option to fill the events with participants.

### Possible Alternatives

In theory, SORMAS data-gen could be used, but that is quite elaborate.

### PR DESCRIPTION
**Pull request by DavidBaldsiefen**
Closes #4185 and #3337 (some other issues regarding the existing case and contact generation were solved too).

Moving the generation to the backend needs to be handled in another issue, as it's not that important at the moment.

### CODE REVIEW

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 1]
```diff
@@ -323,6 +441,18 @@ private VerticalLayout createContactGeneratorLayout() {
...
+		"Party",
+		"Funeral",
+		"Concert",
+		"Fair",
+		"Rallye",
+		"Demonstration",
+		"Football Match",
+		"Tournament",
+		"Festival",
+		"Carinval" };
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:455]: => Carnival

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 2]
```diff
@@ -332,6 +462,12 @@ private static boolean randomPercent(int p) {
 		return random().nextInt(100) <= p;
 	}
 
+	private static int randomInt(int min, int max) {
+		if (max <= min)
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:466]: Use brackets even for one-line statements, please

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 3]
```diff
@@ -508,6 +647,11 @@ private void generateContacts() {
 		ContactGenerationConfig config = contactGeneratorConfigBinder.getBean();
 
 		List<Disease> diseases = config.getDisease() == null ? FacadeProvider.getDiseaseConfigurationFacade().getAllDiseases(true, true, true) : null;
+		Disease disease = config.getDisease();
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:650]: Move this definition one line up, you can then use it as well in the definition of `List<Disease> diseases`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java [hunk 4]
```diff
@@ -803,4 +1087,112 @@ public void setCreateWithVisits(boolean createWithVisits) {
...
+			this.maxContactsPerParticipant = maxContactsPerParticipant;
+		}
+
+		public int getPercentageOfCases() {
+			return percentageOfCases;
+		}
+
+		public void setPercentageOfCases(int percentageOfCases) {
+			this.percentageOfCases = percentageOfCases;
+			if (this.percentageOfCases >= 100)
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/DevModeView.java:1191]: Use brackets

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* The script currently can create cases resulting from events with a disease that differs from the event disease; cases resulting from events can not have a different disease than the event disease; events WITHOUT a disease shouldn't have any resulting cases at all

**Comment by DavidBaldsiefen** [comment]
> The script currently can create cases resulting from events with a disease that differs from the event disease; cases resulting from events can not have a different disease than the event disease; events WITHOUT a disease shouldn't have any resulting cases at all

Thanks for the review! Currently, my intention was that events where no disease is selected will be created with a random disease, and thus all participants and resulting cases will also have that diesease. Would that be ok? (It will still require me to change a line I missed).

Alternatively one could create disease-less events with no cases or contacts

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
I still get cases with diseases that differ from the event diseases. For example, I just generated 10 events with Measles, and the while some of the cases resulting from these events are Measles case, I also have some with Meningitis, Polio, Dengue, Cholera, etc. It seems like the event disease is overridden when the case is actually created.

**Comment by DavidBaldsiefen** [comment]
@MateStrysewske it should work now, thanks for the review!

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**